pkgs/development/tools/misc/clojure-lsp/default.nix

```diff
@@ -1,31 +1,55 @@
-{ lib, stdenv, fetchurl, jre, makeWrapper }:
+{ lib, stdenv, callPackage, fetchurl, fetchFromGitHub, leiningen, openjdk11
+, graalvm11-ce, babashka }:
 
-stdenv.mkDerivation rec {
+let
   pname = "clojure-lsp";
-  version = "2021.02.13-18.04.19";
+  version = "2021.02.14-19.46.47";
+  leiningen11 = leiningen.override ({ jdk = openjdk11; });
 
-  src = fetchurl {
-    url = "https://github.com/clojure-lsp/clojure-lsp/releases/download/${version}/${pname}.jar";
-    sha256 = "sha256-v7+HH77xFxQc0LjVhcrXsdcmmZmrPiMSArxfoOaHkoI=";
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-Zj7/8RcuxCy2xdd+5jeOb1GTsQsX0EVW32k32fA6uf4=";
   };
 
-  dontUnpack = true;
+  repository = callPackage ./repository.nix {
+    inherit src pname version;
+    leiningen = leiningen11;
+  };
+in stdenv.mkDerivation rec {
+  inherit src pname version;
+
+  postPatch = ''
+    # Hack to set maven cache in another directory since MAVEN_OPTS doesn't work
+    substituteInPlace project.clj \
+      --replace ":main" ":local-repo \"${repository}\" :main"
+  '';
+
+  GRAALVM_HOME = graalvm11-ce;
 
-  buildInputs = [ makeWrapper ];
+  buildInputs = [ graalvm11-ce leiningen11 repository ];
+
+  buildPhase = with lib; ''
+    export LEIN_HOME="$(mktemp -d)"
+    bash ./graalvm/native-unix-compile.sh
+  '';
 
   installPhase = ''
-    install -Dm644 $src $out/share/java/${pname}.jar
-    makeWrapper ${jre}/bin/java $out/bin/${pname} \
-      --add-flags "-Xmx2g" \
-      --add-flags "-server" \
-      --add-flags "-jar $out/share/java/${pname}.jar"
+    install -Dm755 ./clojure-lsp $out/bin/clojure-lsp
+  '';
+
+  # TODO: tests are timing out, needs more investigation
+  doCheck = false;
+  checkPhase = ''
+    ${babashka}/bin/bb ${src}/integration-test/run-all.clj ./clojure-lsp
   '';
```

Comment:
Running those tests manually seems to work, so probably there is some other issue.